flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/FailoverRegion.java

```diff
@@ -33,18 +30,21 @@
  */
 public class FailoverRegion {
 
+	/** All vertex IDs in this region. */
+	private final Set<ExecutionVertexID> executionVertexIDs;
+
 	/** All vertices in this region. */
-	private final Map<ExecutionVertexID, FailoverVertex> executionVertices;
+	private final Set<FailoverVertex> executionVertices;
```

Comment:
This is to avoid the cost to convert `Collection<FailoverVertex>` to `Set<FailoverVertex>`. 
We need a `Set<ExecutionVertexID>` and `Set<FailoverVertex>`. But the map provides `Set<ExecutionVertexID>` and `Collection<FailoverVertex>`.

Though I think this may not make a big difference as it is vertex scale complexity.